code-challenges.test.js

```diff
@@ -15,35 +15,113 @@
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+describe("firstRemoveThenShuffle", () => {
+    it("takes in an array, then removes the first item from the array and shuffles the remaining content.", () => {
 
+        const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+        const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 
+      expect(firstRemoveThenShuffle(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+      expect(firstRemoveThenShuffle(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+    })
+  })
 // b) Create the function that makes the test pass.
 
+// make a function that takes in an array
+// use .shift method to remove the first index inside the array
+// make a new variable of array to store the shuffle array
+// make a variable that takes in a random number and use that variable to get the index of the array, add it to the new variable then delete that array by using splice method.
+// repeat the process by using while loop until the array is empty
+// return a shuffle array without the first index
+const firstRemoveThenShuffle =(arr) => {
+  arr.shift()
+  let length = arr.length
+  let randomColors = []
+  let randomNum = 0
+  
+  while (length--){
+      randomNum = Math.floor(Math.random() * (length + 1))
+      randomColors.push(arr[randomNum])
+      arr.splice(randomNum, 1)
+  }
+
+  return randomColors
+}
+
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-const nums2 = [109, 5, 9, 67, 8, 24]
-// Expected output: [5, 109]
+describe("minMax", () => {
+  it("returns an array of the minimum and maximum numbers in that order", () => {
+    
+    const nums1 = [3, 56, 90, -8, 0, 23, 6]
+    // Expected output: [-8, 90]
+    const nums2 = [109, 5, 9, 67, 8, 24]
+    // Expected output: [5, 109]
+     
+    expect(minMax(nums1)).toEqual([-8, 90])
+    expect(minMax(nums2)).toEqual([5, 109])
+  })
+})
 
 
 // b) Create the function that makes the test pass.
 
+// create a function that takes in an aray of numbers
+// create a min and max variable and assign it to the first index on the array
+// use while loop to go through each index in the array
+// use the method max to get the highest number and min to get the minimum number of each array/
+// store them in their new variable, then add the min and max variable to a new array
+// return an array of the minimum and maximum numbers in that order
+const minMax = (arr) => {
```

Comment:
If you are up for the challenge, try making a one-line return statement just using the logic of the Math.max and Math.min. I love that part of coding. After we figure a way to solve the problem, going back to re-evaluate our approach for shorter steps.